scripts/deploy.ts

```diff
@@ -0,0 +1,284 @@
+type Domain = (typeof domains)[number]
+
+const triggerTypeNames = ['block', 'cron', 'event', 'time']
+
+type BlockTrigger = {
+    type: 'block'
+}
+
+type CronTrigger = {
+    type: 'cron'
+    cron: string
+}
+
+type EventTrigger = {
+    type: 'event'
+    filter: {
+        address: string
+        topics: Array<{
+            abiName: string
+            eventName: string
+        }>
+    }
+    blockConfirmations: number
+}
+
+type TimeTrigger = {
+    type: 'time'
+    interval: number
+}
+
+type Trigger = BlockTrigger | CronTrigger | EventTrigger | TimeTrigger
+
+type DeploymentConfig = {
+    domain: Domain
+    args: object
+    secrets: object
+    trigger: Trigger
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//// HELPER FUNCTIONS //////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const createTriggerConfig = (trigger: Trigger): TriggerConfig => {
+    if (trigger.type === 'block') return { type: TriggerType.BLOCK }
+
+    if (trigger.type === 'cron') return { type: TriggerType.CRON, cron: trigger.cron }
+
+    if (trigger.type === 'event') {
+        let topics: Array<string> = []
+
+        for (const topic of trigger.filter.topics) {
+            const abi = JSON.parse(fs.readFileSync(path.join(abisPath, topic.abiName + '.json')))
+            const contractInterface = new utils.Interface(abi)
+            topics.push(contractInterface.getEventTopic(topic.eventName))
+        }
+
+        return {
+            type: TriggerType.EVENT,
+            filter: {
+                address: trigger.filter.address,
+                topics: [topics],
+            },
+            blockConfirmations: Number(trigger.blockConfirmations),
+        }
+    }
+
+    if (trigger.type === 'time') return { type: TriggerType.TIME, interval: Number(trigger.interval) }
+
+    throw new Error('Unknown trigger type')
+}
+
+const retrievePrivateKeyFromKeystore = (keystorePath: string, passwordPath: string): string => {
+    const password = passwordPath ? fs.readFileSync(passwordPath, 'utf8').slice(0, -1) : ''
+    const keystore = fs.readFileSync(keystorePath, 'utf8')
+    return Wallet.fromEncryptedJsonSync(keystore, password).privateKey
+}
+
+const getBufferFromString = (str: string): Buffer => {
+    return Buffer.from(str, 'utf-8')
+}
+
+const createHash = (rawConfig: Buffer, ipfsDeployment: string): string => {
+    const configHash = utils.keccak256(rawConfig)
+    const ipfsDeploymentHash = utils.keccak256(getBufferFromString(ipfsDeployment))
+    return utils.keccak256(getBufferFromString(configHash.concat(ipfsDeploymentHash)))
+}
+
+const pause = () => {
+    return new Promise<void>((resolve) => {
+        prompter.question('Press ENTER to confirm & continue...', () => {
```

Comment:
Need to be able to disable in CI (add flag maybe)

Reply:
done